native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.js

```diff
@@ -158,123 +141,102 @@ const styles = StyleSheet.create({
   },
 });
 
+const ALERT_TYPE_STYLES = {
+  [ALERT_TYPES.SUCCESS]: {
+    iconSource: 'tick-circle',
+    outerStyle: styles.outerContainerSuccess,
+    iconStyle: styles.iconSuccess,
+  },
+  [ALERT_TYPES.WARN]: {
+    iconSource: 'information-circle',
+    outerStyle: styles.outerContainerWarn,
+    iconStyle: styles.iconWarn,
+  },
+  [ALERT_TYPES.ERROR]: {
+    iconSource: 'information-circle',
+    outerStyle: styles.outerContainerError,
+    iconStyle: styles.iconError,
+  },
+  [ALERT_TYPES.NEUTRAL]: {
+    iconSource: 'information-circle',
+    outerStyle: styles.outerContainerNeutral,
+    iconStyle: styles.iconNeutral,
+  },
+};
+
 const BpkBannerAlert = (props) => {
   const {
-    type, message, onAction, dismissable, expanded, actionButtonLabel, style, children, ...rest
+    type,
+    message,
+    onAction,
+    dismissable,
+    expanded,
+    actionButtonLabel,
+    style,
+    children,
+    ...rest
   } = props;
 
   const expandable = children !== null;
 
-  let iconSource = null;
-  let buttonIconSource = null;
-  let iconStyle = null;
-  let buttonIconStyle = null;
-
-  const outerStyleFinal = [styles.outerContainer];
+  const outerStyle = [styles.outerContainer];
+  const contentPaddedStyle = [styles.bannerContainerPadded];
   const expandedChildContainer = [styles.expandedChildContainer];
+  const { iconSource, outerStyle: outerStyleForType, iconStyle } = ALERT_TYPE_STYLES[type] || {};
```

Comment:
I was thinking we default to one of the defined styles i.e SUCCESS or NEUTRAL

Reply:
The required `type` prop will be screaming in the console and i'd rather the component look broken as opposed to "looks ok, lets ignore warning".

If we wanna make it optional then we should put a default prop for `type`

Reply:
Fair point